src/mahoji/lib/abstracted_commands/mahoganyHomesCommand.ts

```diff
@@ -121,15 +134,27 @@ export async function mahoganyHomesBuyCommand(user: MUser, input = '', quantity?
 	return `Successfully purchased ${loot} for ${cost * quantity} Carpenter Points.`;
 }
 
-export async function mahoganyHomesBuildCommand(user: MUser, channelID: string): CommandResponse {
+export async function mahoganyHomesBuildCommand(user: MUser, channelID: string, tier?: string): CommandResponse {
 	if (user.minionIsBusy) return `${user.minionName} is currently busy.`;
 
 	const conLevel = user.skillLevel(SkillsEnum.Construction);
 	const kc = await getMinigameScore(user.id, 'mahogany_homes');
 
+	let tierData = contractTiers.find(contractTier => conLevel >= contractTier.level)!;
+
+	if (tier) {
+		tierData =
+			contractTiers.find(contractTier => tier.toLowerCase() === contractTier.name.toLowerCase()) || tierData;
+		if (!tierData) return 'Error selecting contract tier.';
```

Comment:
Here, and below, if this is a scenario which we think should never happen, just throw an error instead. otherwise, by doing this, you're potentially secretly hiding bugs.